gluten-core/src/main/scala/io/glutenproject/execution/WholeStageTransformer.scala

```diff
@@ -244,26 +244,23 @@ case class WholeStageTransformer(child: SparkPlan, materializeInput: Boolean = f
 
       // If these are two scan transformers, they must have same partitions,
       // otherwise, exchange will be inserted.
-      val allScanPartitions = basicScanExecTransformers.map(_.getPartitions)
-      val allScanPartitionSchemas = basicScanExecTransformers.map(_.getPartitionSchemas)
-      val partitionLength = allScanPartitions.head.size
-      if (allScanPartitions.exists(_.size != partitionLength)) {
+      val allScanLocalFilesNodes = basicScanExecTransformers.map(_.getLocalFilesNodes)
+      val partitionLength = allScanLocalFilesNodes.head.size
+      if (allScanLocalFilesNodes.exists(_.size != partitionLength)) {
```

Comment:
maybe extract a private method to encapsulate this logic that contains `getLocalFilesNodes`,  the assertion of whether the number of partitions is consistent, and the transposition. And doc for this method.